### ISSUE TITLE
Sites with multiple catchments

### ISSUE THREAD

**Issue by philipbaileynar** [namc-utah/namc-bug-db#100]
@Feldspar5 I processed the ShapeFile you gave me on Friday. Generally the polygons look good, but there are 32 sites with duplicate polygons that have different areas.

There are 32 sites with multiple polygons that have different areas (`Station` listed below). Most of the differences are small. But there are sites like `BRUSH_REF` in Oregon that have two records in the ShapeFile that you provided, with each polygon in  a different location.

<img width="591" alt="Screen Shot 2021-03-29 at 10 15 15 AM" src="https://user-images.githubusercontent.com/8645379/112874491-e0aeeb80-9077-11eb-9539-ccfe95d19cd1.png">

I am starting to fall behind, in part because I'm chasing data issues like this. So I will push on and incorporate the data you provided as best I can, just so that I have something to go in the database. But we need to keep a list of "data issues" that NAMC needs to be aware of and resolve before the new system is used in production.

| Station | Count\( \* \) |
| :--- | :--- |
| 0 | 2 |
| BRUSH\_REF | 2 |
| CO-SS-9426 | 2 |
| CY-TR-1028 | 2 |
| CY-TR-1319 | 2 |
| ELKE | 2 |
| G5071 | 2 |
| LASALCK-2 | 2 |
| MB-LS-14866 | 2 |
| MB-RV-1200 | 2 |
| MIDCRMID | 2 |
| MT-SS-13874 | 2 |
| MXR-downstream | 2 |
| ORFLO-1010 | 2 |
| ORSFA-1009 | 2 |
| PALM-07 | 2 |
| PIBO:0123 | 2 |
| PNTH | 2 |
| SF 0.7 | 2 |
| SFREF1 | 2 |
| SW-LS-19727 | 2 |
| SW-SS-23311 | 2 |
| TA-LS-16034 | 2 |
| TA-LS-22178 | 2 |
| TA-LS-22498 | 2 |
| W3012 | 2 |
| W303O2 | 2 |
| W308O2 | 2 |
| W3151 | 2 |
| XE-LS-5023 | 2 |
| XE-RM-5073 | 2 |
| XN-RM-4073 | 2 |

FYI: @kemaier @courtwrightj01 @alherca73

**Comment by philipbaileynar** [comment]
Just adding a screenshot of all polygons for reference:

<img width="1376" alt="Screen Shot 2021-03-29 at 10 31 56 AM" src="https://user-images.githubusercontent.com/8645379/112876178-089f4e80-907a-11eb-8201-412f8a891194.png">

**Comment by philipbaileynar** [comment]
And here's the query for the sites with different areas:

```sql
SELECT Station,
       Count( * )
  FROM (
           SELECT Station,
                  shedareakm,
                  count( * )
             FROM (
                      SELECT IFNULL(Station, ReachID) Station,
                             shedareakm
                        FROM Catchments
                  )
            GROUP BY station,
                     shedareakm
            ORDER BY station
       )
 GROUP BY station
HAVING count( * ) > 1;
```

**Comment by Feldspar5** [comment]
I am sorry to derail you with that shapefile; I suspected that some sitecodes might be incorrect, but had not realized there would duplicate delineations.  I met with Alex and he has passed over a clean shapefile from 2020.  Please use or ignore as is most efficient for you.  I think we are considering creating all new watersheds for a large number of sites (eg. AIM) before we pass into the new db.

**Comment by philipbaileynar** [comment]
Thanks @Feldspar5... The messy data are a minor inconvenience. The API has methods for:

* retrieving catchment polygons
* updating polygons should updates be necessary
* inserting missing polygons when new sites are added (or existing sites require predictors).

My main concern is maintaining confidence in the contents of the new database. I don't want to put in problematic data such that NAMC staff stop trusting the data.

In summary, let's plan a "data verification phase" whereby we collaborate on a series of tasks for verifying the contents of the new database. This is especially important for the areas of the system that were previously not stored in either STREAM or Pilot. Namely, metrics and predictors.

**Comment by philipbaileynar** [comment]
Closing this issue now that we have an API endpoint that allows NAMC to upload catchment polygons.

I have discussed with @Feldspar5 about the longterm workflow for getting catchment polygons into the system. This will probably rely on a NAMC analyst having a ShapeFile of delineated polygons on their computer and then running an R script to load the polygons into the database. The ShapeFile will need the unique NAMC site code as an attribute.
